Adding -c- group: download, collate, upload collated files from Landsat pull

steeleb opened the pull request:
Heya MB!

Lots of fun with multi-GB files in this section, so I added a lot of filtering/mapping functions in the `collate_csvs_from_drive()` function that was originally copied from [the Landsat_C2_SRST template version.](https://github.com/rossyndicate/Landsat_C2_SRST_template/blob/main/data_acquisition/src/collate_csvs_from_drive.R). This creates a bunch more files, but they are more reasonable for R to handle (or any downstream user). 

Primary ask here is general review and style alignment. There are poop ton of conditionals in the `c_collate_Landsat_data/src/collate_csvs_from_drive.R` function, so if you see an opportunity for a re-org there to make the code shorter/more efficient, please say so. Full disclosure, my hope is that I can plug this back into the template repo, so I like retaining the varying options for collation, but if that seems unreasonable and this function is insane, I trust your judgement - just say the word.

I would not advise running this locally. If you want to test any functions, I'm happy to send .rds files to facilitate that, but run time is WAYYYYY too long to run the whole thing IMO. It's mostly a bottleneck at GEE, but the download/collate/upload isn't short either. Please do make sure you can access the files (not signed in as ROSS@gmail) saved in the `c_collate_Landsat_data/out/raw_collated_files_drive_ids_2024-12-16.csv` at a minimum. 

Additional feedback request: What do you think about the proposed walk through in `run_targets.Rmd` that breaks apart the pull/collation groups (a-c) from later (forthcoming) steps? I don't know if this is necessarily an appropriate place for a configuration file specific to run/not run the pull since it would involve a lot of conditionals to the current targets. I'll note that if you think this would be beneficial, now is the optimal time to fix it, since I realized I had an issue with the Landsat pull and need to re-configure and re-run (#27). Another alternative is to let this be the end of the pipeline, and then create a new repo and new pipeline that picks up from here (similar to the download/harmonize vibes from the WQP). I'm all ears here to your thoughts.

Ideal timeline for review is by EOW, but EOD Monday is fine too. I'll be switching back over to siteSR in the meantime, so just LMK if that does/doesn't work for your current tasks.

Progress on #5

mbrousil commented on c_collate_Landsat_data/src/download_csvs_from_drive.R:
```diff
@@ -0,0 +1,63 @@
+#' to the untracked `c_collate_Landsat_data/down/` folder
+#'
+#' @param file_type text string; unique string for filtering files to be 
+#' downloaded from Drive. Some options: "457" (Landsat 4, 5, 7), "metadata", 
```
question: Would it make sense to standardize these with, e.g., the target `c_data_segments`? Also what would happen if someone used just "4" or something instead of "457"? Not sure if that's an intended use or not

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+  if (length(metadata) > 0) {
+    
+    # process LS457 and LS89 mission groups separately
+    misson_groups <- c("LS457", "LS89")
+    walk(misson_groups,
+         .f = \(mg) {
```
```suggestion
    mission_groups <- c("LS457", "LS89")
    walk(mission_groups,
         .f = \(mg) {
```

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+    # process LS457 and LS89 mission groups separately
+    if (is.null(file_type)) {
+      
+      misson_groups <- c("LS457", "LS89")
+      
+      walk(misson_groups,
+           .f = \(mg) {
```
```suggestion
      mission_groups <- c("LS457", "LS89")
      
      walk(mission_groups,
           .f = \(mg) {
```

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+                                             # get column names that need to be 
+                                             # coerced to numeric (all but index)
+                                             df_names <- names(df)[2:length(names(df))]
+                                             # corece columns to numeric and add
```
```suggestion
                                             # coerce columns to numeric and add
```

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+                                  # get column names that need to be 
+                                  # coerced to numeric (all but index)
+                                  df_names <- names(df)[2:length(names(df))]
+                                  # corece columns to numeric and add
```
```suggestion
                                  # coerce columns to numeric and add
```

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+                                     # get column names that need to be 
+                                     # coerced to numeric (all but index)
+                                     df_names <- names(df)[2:length(names(df))]
+                                     # corece columns to numeric and add
```
```suggestion
                                     # coerce columns to numeric and add
```

mbrousil commented on c_collate_Landsat_data/src/collate_csvs_from_drive.R:
```diff
@@ -0,0 +1,354 @@
+                           # get column names that need to be 
+                           # coerced to numeric (all but index)
+                           df_names <- names(df)[2:length(names(df))]
+                           # corece columns to numeric and add
```
```suggestion
                           # coerce columns to numeric and add
```

mbrousil reviewed (approved):
Hey B,

This looks good to me! I didn't see any major things that I felt needed to be changed. Only a couple of small things that I pointed out. I don't think I would have done the collate function much differently from what you did, so hopefully that's a good thing to hear haha. Drive file successfully accessed with drive_get(as_id()). The walkthrough in the Rmd also makes sense. I don't think that I would go the route of a config file for that and for now a single repo seems fine to me. In the future you could split it up if you felt that it made more sense once those other steps are written up?

steeleb commented:
Thanks @mbrousil! I addressed the download filter (and added the same warning to the collate filter) - so basically, if you use the 'file_type' argument and it is not LS457, LS89, metadata, or pekel you'll get a warning.
